lib/Service/DocumentService.php

```diff
@@ -422,11 +422,12 @@ public function resetDocument(int $documentId, bool $force = false): void {
 
 			$this->unlock($documentId);
 
-			$this->stepMapper->deleteAll($documentId);
+			$this->stepMapper->deleteBeforeVersion($documentId, $document->getLastSavedVersion());
```

Comment:
This should still be fine as a cleanup as all steps in the saved version are also persisted in the y.js state, right @max-nextcloud ?

Reply:
I actually changed the logic of `resetDocument()` in #5481 to always delete all session data: document, sessions and steps from database as well as yjs document state file. And that will happen in the following scenarios:

* A document gets changed from outside text and there's no unsaved steps
* Cleanup background job runs for a stale document session (and no unsaved steps)
* occ command `text:reset` is run

I wonder whether the second case should just remove sessions and steps instead.